src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/GMMVariantTrainArgumentCollection.java

```diff
@@ -0,0 +1,180 @@
+     */
+    @Advanced
+    @Argument(fullName = "maximum-training-variants", doc = "Maximum number of training data", optional = true)
+    protected int MAX_NUM_TRAINING_DATA = 2500000;
+
+    /**
+     * MQ is capped at a "max" value (60 for bwa-mem) when the alignment is considered perfect. Typically, a huge
+     * proportion of the reads in a dataset are perfectly mapped, which yields a distribution of MQ values with a
+     * blob below the max value and a huge peak at the max value. This does not conform to the expectations of the
+     * Gaussian mixture model of VQSR and has been observed to yield a ROC curve with a jump.
+     *
+     * This argument aims to mitigate this problem. Using MQCap = X has 2 effects:  (1) MQs are transformed by a scaled
+     * logit on [0,X] (+ epsilon to avoid division by zero) to make the blob more Gaussian-like and (2) the transformed
+     * MQ=X are jittered to break the peak into a narrow Gaussian.
+     *
+     * If this option is not used, or if MQCap is set to 0, MQ will not be transformed.
+     * Note that AS_MQ uses --mq-jitter instead to control artificial variance
+     */
+    @Advanced
+    @Argument(fullName="mq-cap-for-logit-jitter-transform", shortName = "mq-cap", doc="Apply logit transform and jitter to MQ values", optional=true)
+    public int MQ_CAP = 0;
+
+    /**
+     * Amount of jitter (as a multiplier to a Normal(0,1) distribution) to add to the AS_MQ and transformed MQ values
+     */
+    @Advanced
+    @Argument(fullName="mq-jitter", doc="Amount of jitter (as a multiplier to a Normal(0,1) distribution) to add to the AS_MQ and transformed MQ values", optional = true)
+    public double MQ_JITTER = 0.05;
+
+    @Advanced
+    @Argument(fullName = "debug-stdev-thresholding", doc="Output variants that fail standard deviation thresholding to the log for debugging purposes. Redirection of stdout to a file is recommended.", optional = true)
+    public boolean debugStdevThresholding = false;
+
+    Hyperparameters hyperparameters = null;
+
+    void readHyperparameters() {
+        IOUtils.canReadFile(hyperparametersJSONFile);
+        try {
+            final ObjectMapper mapper = new ObjectMapper();
+            hyperparameters = mapper.readValue(hyperparametersJSONFile, Hyperparameters.class);
+        } catch (final Exception e) {
+            throw new UserException.BadInput("Could not read hyperparameters JSON.", e);
+        }
+    }
+
+    // TODO perhaps just put this in the BGMM classes?
+    // there is the complication of specifying mean and covariance priors differently here,
+    // as well as the possible desire to have different defaults;
+    // would also require slightly more code to invoke a BGMM if we use a builder for these instead
+    static final class Hyperparameters {
+        @JsonProperty("n_components")
+        int nComponents = 6;
+        @JsonProperty("tol")
+        double tol = 1.;
+        @JsonProperty("reg_covar")
```

Comment:
These argument names intentionally match the ones in the sklearn implementation, so we can just reference their documentation in some way. But it is indeed still an open question in my mind what the best way to document this in our infrastructure might be, especially if the intent is to allow an open ended design so that any hyperparameters can be passed!